src/globus_sdk/utils.py

```diff
@@ -276,3 +278,87 @@ def classproperty(func: t.Callable[[T], R]) -> _classproperty[T, R]:
     def classproperty(func: t.Callable[[T], R]) -> _classproperty[T, R]:
         # type cast to convert instance method to class method
         return _classproperty(t.cast(t.Callable[[t.Type[T]], R], func))
+
+
+@functools.lru_cache(maxsize=None)
+def read_sphinx_params(docstring: str) -> tuple[str, ...]:
+    """
+    Given a docstring, extract the `:param:` declarations into a tuple of strings.
+
+    :param docstring: The ``__doc__`` to parse, as it appeared on the original object
+
+    Params start with `:param ...` and end
+    - at the end of the string
+    - at the next param
+    - when a non-indented, non-param line is found
+
+    Whitespace lines within a param doc are supported.
+
+    All produced param strings are dedented.
+    """
+    docstring = textwrap.dedent(docstring)
+
+    result: list[str] = []
+    current: list[str] = []
+    for line in docstring.splitlines():
+        if not current:
+            if line.startswith(":param"):
```

Comment:
Surely sphinx loads these as a part of its docstring parsing these. Do we have any mechanism to tap into that in a plugin instead of doing it ourselves?

Reply:
I don't think there's an easy way to do this via sphinx. I'm sure the info is in there, but complex sphinx extensions are pretty hard to write. 

I also like that this impacts `help()` in the REPL and maybe some IDEs as well. But I'm not overly attached to that. 

We could rewrite from a decorator to a special sphinx directive in the docstring. That kind of extension is on the easy side to write. But I'd stick with parsing the data ourselves -- I think it's actually a lot easier than getting that data from sphinx, partly because it introduces a requirement that we get the ordering of events correct. (Sphinx parse has to be done before we can access it.)

Reply:
Yeah, per our out-of-band discussion I want to refactor the sphinx extension into a subpackage. So I agree that this should have a better home -- and that could lead to us being able to write some unit tests for our sphinx extension, which I would very much like.